internals/cli/env.go

```diff
@@ -266,3 +433,99 @@ func splitVar(prefix, separator, envVar string) (string, string, bool) {
 	prefix = fmt.Sprintf("%s%s", strings.ToUpper(prefix), separator)
 	return split[0], split[1], strings.HasPrefix(strings.ToUpper(split[0]), prefix)
 }
+
+type Argument struct {
+	Store       ArgValue
+	Name        string
+	Required    bool
+	Placeholder string
+	Description string
+	Hidden      bool
+}
+
+type ArgValue interface {
+	Set(string) error
+}
+
+func ArgumentRegister(params []Argument, args []string) error {
+	for i, arg := range args {
+		err := params[i].Store.Set(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ArgumentArrRegister(params []Argument, args []string) error {
+	for _, arg := range args {
+		err := params[0].Store.Set(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type StringArgValue struct {
+	Param string
+}
```

Comment:
`StringArrArgValue` cannot be embedded, since it has an array as a parameter. Arrays cannot be embedded, they are non-defined non-pointer types.
`StringArgValue` cannot be embedded, because it has an unexported type inside as the basic element (string).

Reply:
That's alright. We can leave those as is.